src/libraries/Common/src/System/Net/MultiArrayBuffer.cs

```diff
@@ -0,0 +1,431 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Net
+{
+    // Warning: Mutable struct!
+    // The purpose of this struct is to simplify buffer management in cases where the size of the buffer may grow large (e.g. >64K),
+    // thus making it worthwhile to add the overhead involved in managing multiple individual array allocations.
+    // Like ArrayBuffer, this manages a sliding buffer where bytes can be added at the end and removed at the beginning.
+    // Unlike ArrayBuffer, the buffer itself is managed using 16K blocks which are added/removed to the block list as necessary.
+
+    // [ActiveBuffer] contains the current buffer contents; these bytes will be preserved on any call to TryEnsureAvailableBytesUpToLimit.
+    // [AvailableBuffer] contains the available bytes past the end of the current content,
```

Comment:
Nit: rather than using braces like `[ActiveBuffer]`, how about using single quotes or backticks, e.g. `'ActiveBuffer'`?  The braces just immediately make me think we're talking about attributes.

Reply:
Sure